File: 프로그래머스/lv0/120886. A로 B 만들기/A로 B 만들기.java
import java.util.Arrays;

class Solution {
    public int solution(String before, String after) {
        int answer = 0;
        
        char[] a = after.toCharArray();
		char[] b = before.toCharArray();
		// after와 before를 array로 변환하고
        
		Arrays.sort(a);
		Arrays.sort(b);
		// 각 array를 정렬한다
        
		String aa = Arrays.toString(a);
		String bb = Arrays.toString(b);
		// 정렬한 array를 다시 string으로 변환해준다
        
		if(aa.equals(bb)) {
			answer = 1;
		} // 이때, 변환된 before에 after이 포함되면
          // before로 after를 만들 수 있다는 의미로 answer에 1을 넣어주고
          // 이에 해당하지 않으면 0인 상태로 두면 안된다
        
        return answer;
    }
}

File: 프로그래머스/0/120886. A로 B 만들기/A로 B 만들기.js
function solution(before, after) {
    let answer = 0;
    
    let a = after.split('');
    let b = before.split('');
    // 문자열을 배열로 변환
    
    a.sort();
    b.sort();
    // 배열 정렬
    
    let aa = a.join('');
    let bb = b.join('');
    // 정렬된 배열을 다시 문자열로 변환
    
    if (aa === bb) {
        answer = 1;
    } // 정렬된 문자열을 비교하여 같다면 answer에 1을 할당
    
    return answer;
}
